src/export/ValueSetExporter.ts

```diff
@@ -130,6 +135,155 @@ export class ValueSetExporter {
     }
   }
 
+  private performExpansion(valueSet: ValueSet) {
+    if (
+      valueSet.expansion?.parameter?.find(
+        parameter => parameter.name === 'sushi-generated' && parameter.valueBoolean === true
+      )
+    ) {
+      const errors: string[] = [];
+      // the compose property must be defined
+      if (!valueSet.compose) {
+        errors.push('No composition defined.');
+      }
+      // filter operators are prohibited
+      if (
+        [...(valueSet.compose?.include ?? []), ...(valueSet.compose?.exclude ?? [])].some(
+          compose => {
+            return compose.filter?.some(filter => {
+              return filter.op?.length;
+            });
+          }
+        )
+      ) {
+        errors.push('Composition contains filter operators.');
+      }
+      // including/excluding value sets are prohibited
+      if (
+        [...(valueSet.compose?.include ?? []), ...(valueSet.compose?.exclude ?? [])].some(
+          compose => {
+            return compose.valueSet?.length && !compose.concept?.length;
+          }
+        )
+      ) {
+        errors.push('Composition contains other value sets.');
+      }
+      // full usage of non-local code systems are prohibited
+      const referencedCodeSystems = fromPairs(
+        [...(valueSet.compose?.include ?? []), ...(valueSet.compose?.exclude ?? [])]
+          .filter(compose => {
+            return !compose.concept?.length && compose.system;
+          })
+          .map(compose => [
+            compose.system,
+            this.fisher.fishForFHIR(compose.system, Type.CodeSystem)
+          ])
+      );
+      // if we were able to fish up definitions for all of these systems, we're good
+      if (Object.values(referencedCodeSystems).includes(undefined)) {
+        errors.push('Composition contains code systems without available concept lists.');
+      }
+
```

Comment:
Oh.  And last thing.  What if the ValueSet actually specified the `version` as well?  Should we be checking to ensure the `version` in the fished up `CodeSystem` matches the `version` specified in the composition?

Reply:
Checking the version seems correct to me. This has led me to a question about handling specified versions. When fishing up metadata for the CodeSystem in `ValueSetExporter.setCompose`, the canonical url may include a version. What should be done if the version specified in the rule is not the same as the version in the canonical url? Is it immediately an error, or should the exporter prefer the version specified in the rule? The ValueSet may not be getting expanded, so it may not matter if the full definition for that version of the CodeSystem can be found later on.

This has also revealed what appears to be a bug in the `setCompose` method, in that it appears to always fail if you specify the compose rule with the CodeSystem name and version, like so:
```
* include codes from system AllergyIntoleranceClinicalStatusCodes|4.0.1
```
This seems like something that ought to work, and it's also interfering with this feature, so it will be fixed as part of this PR.

Reply:
Why do we want to include CodeSystems with `CodeSystem.content` set to `example`. The definition for `example` is:
> A few representative concepts are included in the code system resource. There is no useful intent in the subset choice and there's no process to make it workable: it's not intended to be workable.

This doesn't sound like it would be useful for us to expand.

Reply:
As discussed in the standup, you're right @ngfreiter -- we probably should not have included `example`.  I was assuming `example` meant an example (but complete) `CodeSystem`; I did not realize it could mean a few "example" codes from a real `CodeSystem`.  I'm not sure exactly how that differs from `fragment`, but I think we should play it safe and only include `complete` `CodeSystem`s then!  Thanks for reading the definitions!